### ISSUE TITLE
Auto Closing Brackets and Quotes

### ISSUE THREAD

**Issue by Sushi2k6** [thonny/thonny#1942]
I have been using thonny for the past 3 months, and I'd give it a 4/5 (according to me). I want my IDE (or editor) to be simple, accessible and quick, basically it should be **lightweigh**t. An that's what thonny is. But with a few setbacks, like

-> Auto closing Brackets and Quotes
-> Ability to install packages from different sources
->Better Assistant

If these are implemented in thonny, It'd be a lot better for everyone
Hope it will. (cuz I cant find anything that suites me, and the closest one is thonny!)

**Comment by aivarannamaa** [comment]
Thank you for the suggestions!

Auto-closing parens in in my TODO list: #59

> Ability to install packages from different sources

Do you mean specifying different or extra index instead or besides PyPI?

> Better Assistant

What do you mean specifically?

**Comment by Sushi2k6** [comment]
Besides PyPI.
And, Assistant can give definitions, indicate errors on code while typing, this makes it so much easier.

BTW, The closing parens doesn't include [] and {} right? It'd be better if the quote was also closed automatically

**Comment by ser9ee** [comment]
> Thank you for the suggestions!
> 
> Auto-closing parens in in my TODO list: #59
> 
> > Ability to install packages from different sources
> 
> Do you mean specifying different or extra index instead or besides PyPI?
> 
> > Better Assistant
> 
> What do you mean specifically?

Hello.
Thonny is very good and i like it so much, but auto-closing is soooo important for usability. please add this option as quickly as possible. Thank you for your great application!
(), [], {}, '', ""

**Comment by Sushi2k6** [comment]
Hello! Is the issue solved yet? It hasn't been added yet.

**Comment by aivarannamaa** [comment]
No, it's still in the TODO list and unfortunately there are lots of more important issue before this.

**Comment by Sushi2k6** [comment]
oh ok. Is there any possible way to see the TODO list?

**Comment by aivarannamaa** [comment]
Basically https://github.com/thonny/thonny/issues is the TODO list. You can filter by the milestone to get a more structured picture.

**Comment by Sushi2k6** [comment]
Oh ok, thanks. Actually I was confused by this 
![image](https://user-images.githubusercontent.com/89247190/159872035-34d9fefc-5bee-4a02-94d5-8d008a4f73b7.png)

**Comment by RushilJalal** [comment]
When is the autoclosing quotes and brackets going to be implemented? Thanks

**Comment by aivarannamaa** [comment]
> When is the autoclosing quotes and brackets going to be implemented?

I can't promise any date. When more import issues are fixed and I have time for this.

**Comment by visualearner** [comment]
Hello, does Thonny have Auto Closing Brackets and Quotes yet? I haven't been able to find an answer but it would be so much smoother if it had it! thank you
